runtime/gc_vlhgc/MemorySubSpaceTarok.cpp

```diff
@@ -1355,18 +1385,169 @@ MM_MemorySubSpaceTarok::calculateExpandSize(MM_EnvironmentBase *env, UDATA bytes
+			/* The heap size we tested will give us an acceptable amount of free space, and better gc cpu % */
+			reccomendedHeapSize += suggestedChange;
+			foundAcceptableHeapSizeChange = true;
+			Trc_MM_MemorySubSpaceTarok_getHeapSizeWithinBounds_1(env->getLanguageVMThread(), reccomendedHeapSize, potentialHybridOverhead);
+		} else {
+			/* The heap size we tried was not satisfactory. Keep searching */
+			suggestedChange += heapSizeChangeGranularity;
+		}
+	}
+
+	/* 
+	 * If looking at the overhead curve did not work, base the expansion/contraction off of how high/low the hybrid overhead is. 
+	 * For each % above heapExpansionGCTimeThreshold, expand more aggresively. 
+	 * Ie, if heapExpansionGCTimeThreshold = 13, and measured hybrid overhead is 20, we want to expand by a larger amount than if measured hybrid overhead is 14.
+	 */
+	if (!foundAcceptableHeapSizeChange) {
+		/* Expansion and contraction may increase/decrease by larger/smaller amounts */
+		uintptr_t sizeChangeFactor = 0;
+
+		/* percentDiff is represented as percent between 0 - 100 */
+		double percentDiff = 0.0;
+
+		if (currentHybridHeapScore >= (double)_extensions->heapExpansionGCTimeThreshold) {
+			/* Try to bring hybridHeapScore a little bit below _extensions->heapExpansionGCTimeThreshold */
+			percentDiff = currentHybridHeapScore - (double)_extensions->heapExpansionGCTimeThreshold;
+
+			/* Limit the percent difference to prevent any accidental big changes. 
+			 * This helps deal with cases with one GC that is too long for whatever reason, causing a massive, undesired increase in heap size 
+			 */
+			percentDiff = OMR_MAX(percentDiff, 5.0);
+
+			/* Be a bit more aggressive with expansion than contraction.  */	
+			sizeChangeFactor = 2;	
+
+		} else if (currentHybridHeapScore <= (double)_extensions->heapContractionGCTimeThreshold) {
+			/* Try to bring hybridHeapScore a little bit above _extensions->heapContractionGCTimeThreshold.
+			 * If this doesn't cause the hybrid heap score to increase enough so that it is within acceptable bounds, this path will be taken 
+			 * again later, and another small contraction will occur.
+			 */
+			percentDiff = currentHybridHeapScore - (double)_extensions->heapContractionGCTimeThreshold;
+			sizeChangeFactor = 1;
+		} 
+
+		/* Since percentDiff is 0 - 100, make sure to convert it to 0-1 percentage */
+		double heapSizePercentChange = (1.0 + ((double)sizeChangeFactor * (percentDiff / 100 )));
+		Trc_MM_MemorySubSpaceTarok_getHeapSizeWithinBounds_2(env->getLanguageVMThread(), heapSizePercentChange);
+		reccomendedHeapSize = (uintptr_t)(heapSizePercentChange * reccomendedHeapSize);
+	}
+
+	return reccomendedHeapSize;
+}
+
+
+double 
+MM_MemorySubSpaceTarok::calculateGcPctForHeapChange(MM_EnvironmentBase *env, IDATA heapSizeChange)
+{
+	MM_EnvironmentVLHGC *envVLHGC = (MM_EnvironmentVLHGC *)env;
```

Comment:
we typically name the one from the param envBase, and then this one just env

Reply:
You are right the existing pattern is predominant. I like what I proposed because the sites that use them will have a bit less characters, but it's not a strong argument. We can leave it as is - up to you.